inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MongoDBReader.java

```diff
@@ -0,0 +1,326 @@
+        // snapshot info
+        snapshot = new MongoDBSnapshotBase(offsetStoreFileName);
+        String offset = jobConf.get(JOB_DATABASE_OFFSETS, "");
+        snapshot.save(offset, new File(offsetStoreFileName));
+        // offset info
+        specificOffsetFile = jobConf.get(JOB_DATABASE_OFFSET_SPECIFIC_OFFSET_FILE, "");
+        specificOffsetPos = jobConf.get(JOB_DATABASE_OFFSET_SPECIFIC_OFFSET_POS, "-1");
+    }
+
+    /**
+     * start the embedded debezium engine
+     *
+     * @param jobConf job conf
+     */
+    private void startEmbeddedDebeziumEngine(JobProfile jobConf) {
+        DebeziumEngine<ChangeEvent<String, String>> debeziumEngine = DebeziumEngine.create(Json.class)
+                .using(this.buildMongoConnectorConfig(jobConf))
+                .notifying(this::handleChangeEvent)
+                .using(this::handle)
+                .build();
+        this.executor = Executors.newSingleThreadExecutor();
+        this.executor.execute(debeziumEngine);
+    }
+
+    /**
+     * Handle the completion of the embedded connector engine.
+     *
+     * @param success {@code true} if the connector completed normally,
+     *                or {@code false} if the connector produced an error
+     *                that prevented startup or premature termination.
+     * @param message the completion message; never null
+     * @param error   the error, or null if there was no exception
+     */
+    private void handle(boolean success, String message, Throwable error) {
+        //jobConf.getInstanceId()
+        if (!success) {
+            LOGGER.error("{}, {}", message, error);
+        }
+    }
+
+    /**
+     * A Configuration object is basically a decorator around a {@link Properties} object.
+     *
+     * @return Configuration
+     */
+    private Properties buildMongoConnectorConfig(JobProfile jobConf) {
+        Configuration.Builder builder = Configuration.create();
+        MongoDbConnectorConfig.ALL_FIELDS
+                .forEach(field -> {
+                            String defaultValue = field.defaultValueAsString();
+                            String value = jobConf.get(JOB_PARAM_PREFIX + field.name(), defaultValue);
+                            // Configuration parameters are not set and there is no default value
+                            if (StringUtils.isBlank(defaultValue) && StringUtils.isBlank(value)) {
+                                return;
+                            }
+                            builder.with(field, value);
+                        }
+                );
+        Properties props = builder.build().asProperties();
+        props.setProperty("offset.storage.file.filename", offsetStoreFileName);
+        props.setProperty("connector.class", MongoDbConnector.class.getCanonicalName());
+        props.setProperty("name", instanceId);
+
+        String snapshotMode = props.getOrDefault(MongoDbConnectorConfig.SNAPSHOT_MODE.name(), "").toString();
```

Comment:
It is recommended to change the Key to job.mongoJob.snapshot.mode, and use it as a constant, because the mode was passed to the meeting in the previous design job